aerogear/mobile-security-service
# Added a custom middleware handler to log all incoming requests

## Motivation

JIRA: https://issues.jboss.org/browse/AEROGEAR-8450

## What

Added a custom middleware handler to `middleware.go` that will log (using logrus) certain attributes of every incoming requests. Those attributes are:

- Request method
- Request path
- Client IP Address
- Response time

## Why

So that we can monitor requests made to the server. We could have used default request logger that comes with Echo, but since we are using logrus in other places it is best to stick to one logger.

## How

Referred to the [official docs on how to write custom middleware](https://echo.labstack.com/cookbook/middleware) for Echo.

## Verification Steps
 
1. Run the application:

```sh
go run cmd/mobile-security-service/main.go
```

2. In another terminal window, make a HTTP request to the API:

```sh
> curl --request GET http://localhost:3000/
{"message":"Not Found"}
```

3. Check the first terminal window again, you should see the request info logged like so:

```sh
⇨ http server started on [::]:3000
time="2019-01-29T16:07:18Z" level=info client_ip="[::1]:48274" method=GET path=/ response_time=80ns
```

## Checklist:


## Progress
